### ISSUE TITLE
Cannot validate zero numeric keys using Key or KeyNested

### ISSUE THREAD

**Issue by alganet** [Respect/Validation#587]
```php
--TEST--
keyNested()
--FILE--
<?php
require 'vendor/autoload.php';

use Respect\Validation\Validator as v;

$array = [
    0 => [
        0 => 123
    ]
];


var_dump(v::keyNested('0.0', v::numeric())->validate($array));

?>
--EXPECTF--
bool(true)

```

More details soon.

**Comment by henriquemoody** [comment]
That's because it uses `isset()` instead of `array_key_exists()` I think: https://github.com/Respect/Validation/blob/master/library/Rules/KeyNested.php#L40

**Comment by alganet** [comment]
@henriquemoody yep, that is probably part of the problem. I haven't found a fix yet though, something else is also interfering with it. On the `Key` rule is the `empty()` check on the constructor.

**Comment by alganet** [comment]
Fixed by 84fa389d4

**Comment by henriquemoody** [comment]
\o/
